lib/tasks/integration.rake

```diff
@@ -1,144 +1,218 @@
+# frozen_string_literal: true
+
 require 'color'
+require 'benchmark'
+
 
-desc 'Run continuous integration'
-task :integrate, :log do |_, args|
-  if ENV['CI']
-    ENV['COVERAGE'] = '1'
-    ENV['PERCY_ENABLE'] = '0' # percy will be enabled just for one task
+def print_banner_around
+  lambda do |line|
+    puts
+    puts "=" * 40
+    puts "== #{Color::BOLD}#{line}#{Color::CLEAR}"
+    puts "=" * 40
+    puts
+
+    line
   end
+end
+
+def resolve_test_directories
```

Comment:
Test directory should be defined as a folder that has one or more `*_test.rb` files. 
I think there should be no need for whitelists/blacklists.


```ruby
root = Pathname('test')
test_files = Pathname.glob('test/**/*_test.rb')
test_groups = test_files.group_by { |path|  path.first.descend.take(2).last }
test_groups.keys # => [#<Pathname:test/unit>,
 #<Pathname:test/mailers>,
 #<Pathname:test/cve>,
 #<Pathname:test/decorators>,
 #<Pathname:test/date_parsing_test.rb>,
 #<Pathname:test/integration>,
 #<Pathname:test/models>,
 #<Pathname:test/support>,
 #<Pathname:test/workers>,
 #<Pathname:test/subscribers>,
 #<Pathname:test/events>,
 #<Pathname:test/performance>,
 #<Pathname:test/controllers>,
 #<Pathname:test/helpers>,
 #<Pathname:test/functional>,
 #<Pathname:test/factories_test.rb>,
 #<Pathname:test/remote>]
```

Reply:
i like your approach here @mikz and I'd like to go with it: 

I've tried this out locally and I see we get 3 extra 

```

Reply:
I know. That was kind of my point. 

We need to define where to run them or if to move them (for example to unit tests).